service/verify.go

```diff
@@ -12,7 +14,7 @@ import (
 type (
 	// VerifiableRequest adds possibility to sign and verify request header.
 	VerifiableRequest interface {
-		Marshal() ([]byte, error)
+		MarshalTo([]byte) (int, error)
```

Comment:
What does this function do? Does it panic or append if provided slice has insufficient length? Anyway I believe it is better to have `Marshal()` as well (it could be defined via `MarshalTo()`)

Reply:
Discussed. We adds `Size() int` to interface and checks that current slice of bytes from pool, enough for marshall message, if not, reallocate more slice of bytes. By default, we use 4.5Mb